### ISSUE TITLE
[REVIEW]: Computer-Aided Generation of N-shift RWS

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3431]
**Submitting author:** @benjaminbolling (<a href="http://orcid.org/0000-0002-6650-5365">Benjamin Edward Bolling</a>)
**Repository:** https://github.com/benjaminbolling/RSW
**Branch with paper.md** (empty if default branch): 
**Version:** 1.0.0
**Editor:** @majensen
**Reviewers:** @ShantanuDash, @magedhelmy1
**Archive:** 10.5281/zenodo.7618915

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/b41d06d174b2f0692ed9ac06d745a895/status.svg)](https://joss.theoj.org/papers/b41d06d174b2f0692ed9ac06d745a895)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/b41d06d174b2f0692ed9ac06d745a895"><img src="https://joss.theoj.org/papers/b41d06d174b2f0692ed9ac06d745a895/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/b41d06d174b2f0692ed9ac06d745a895/status.svg)](https://joss.theoj.org/papers/b41d06d174b2f0692ed9ac06d745a895)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/benjaminbolling/RSW" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@magedhelmy1 & @ShantanuDash, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @majensen know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/benjaminbolling/RSW">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@benjaminbolling) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/benjaminbolling/RSW">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@benjaminbolling) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @magedhelmy1, @ShantanuDash it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Failed to discover a `Statement of need` section in paper

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.03 s (286.6 files/s, 66267.3 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                           5             56            124           1806
Markdown                         3             71              0            223
TeX                              1              2              0             22
YAML                             1              0              0              8
-------------------------------------------------------------------------------
SUM:                            10            129            124           2059
-------------------------------------------------------------------------------


Statistical information for the repository '4b552e918940271eb0edc6dd' was
gathered on 2021/06/29.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
benjaminbolling                 32          4373           2387          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
benjaminbolling            1986           45.4          3.6                6.24
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/s0166-218x(01)00258-x is OK

MISSING DOIs

- 10.1201/9781584889304-33 may be a valid DOI for title: Python Language Reference

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03431/joss.03431/10.21105.joss.03431.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03431/joss.03431/10.21105.joss.03431.pdf) :page_facing_up: :point_left:

**Comment by benjaminbolling** [comment]
Dear @majensen, @magedhelmy1 and @ShantanuDash, I will update the document to provide a more clearer statement of need. Anything else you need from me to get going?

**Comment by magedhelmy1** [comment]
> Dear @majensen, @magedhelmy1 and @ShantanuDash, I will update the document to provide a more clearer statement of need. Anything else you need from me to get going?

Thanks. I will take a look at it and come back with my comments

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03431/joss.03431/10.21105.joss.03431.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03431/joss.03431/10.21105.joss.03431.pdf) :page_facing_up: :point_left:

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03431/joss.03431/10.21105.joss.03431.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03431/joss.03431/10.21105.joss.03431.pdf) :page_facing_up: :point_left:

**Comment by benjaminbolling** [comment]
Ready for now. Thanks all!

**Comment by ShantanuDash** [comment]
@majensen could you resend the invitation again. I forgot to join.

**Comment by majensen** [comment]
@ShantanuDash no problem

**Comment by majensen** [comment]
@whedon re-invite @ShantanuDash as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@shantanudash please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by whedon** [comment]
:wave: @ShantanuDash, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @magedhelmy1, please update us on how your review is going (this is an automated reminder).

**Comment by benjaminbolling** [comment]
Issues commented, close if ready.

**Comment by magedhelmy1** [comment]
@whedon re-invite @magedhelmy1 as reviewer

**Comment by whedon** [comment]
I'm sorry @magedhelmy1, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by magedhelmy1** [comment]
@majensen kindly reinvite me :)

**Comment by danielskatz** [comment]
@whedon re-invite @magedhelmy1 as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@magedhelmy1 please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by majensen** [comment]
thanks @danielskatz

**Comment by majensen** [comment]
@ShantanuDash @magedhelmy1 I'd like to restart this review process-- can I ask you about progress? Thanks.

**Comment by ShantanuDash** [comment]
@majensen as I told @benjaminbolling everything was fine except me and @magedhelmy1 asked for some additions to be done in the paper.

**Comment by benjaminbolling** [comment]
Dear all, sorry for the delay - I will address and push tomorrow. Promise!

**Comment by benjaminbolling** [comment]
Hi, almost there - just need 1 more day and I will push it to git.

**Comment by majensen** [comment]
@benjaminbolling - how is this effort coming along?

**Comment by benjaminbolling** [comment]
@majensen : Thanks for the reminder/motivator. I push it now.

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03431/joss.03431/10.21105.joss.03431.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03431/joss.03431/10.21105.joss.03431.pdf) :page_facing_up: :point_left:

**Comment by benjaminbolling** [comment]
@magedhelmy1 and @ShantanuDash : How is the review going? 🙂

**Comment by magedhelmy1** [comment]
Hi Benjamin, to be honest, I did not have the time to look at it again because when the gaps between your reviews are longer than 2months+ I have to re-read the whole paper again and remind myself of what was it about.
As you can see, you mentioned on "Sep 23" you will do the reviews, but did not do it until mid-November which is quite a significant gap.
Anyways, I will try to look at it in January and come back with feedback, it will be hard to find time before Christmas.

**Comment by benjaminbolling** [comment]
Sorry again for the delay, the only excuse there is that I had lots of work and was moving. Thanks!

**Comment by magedhelmy1** [comment]
No worries :) All the best in your work and looking forward to reading it :)

**Comment by majensen** [comment]
Happy new year all - let's try to make the big push over the line in January!

**Comment by magedhelmy1** [comment]
Happy New Year to you too! Yes, I agree. Hopefully, I will have my comments ready by the end of this week.

**Comment by benjaminbolling** [comment]
Thanks and happy new year to you all!

**Comment by magedhelmy1** [comment]
Hi Benjamin, thanks for updating the paper.
It looks much better now.

I have now ticked the parts related to the paper.

- Reference 
- Summary
- A statement of need
- State of the field
---------------------------
I have follow-up feedback/questions.
1. Functionality: Have the functional claims of the software been confirmed?

1.1 - I am finding it hard to follow the software's functionality from just screenshots and captions. Can you please include a [user flow](https://userguiding.com/blog/ux-user-flow-diagram/) of all the scenarios possible with your software, so it is clear how to use it? There is no one way of doing a user flow so, please do whatever you are most confident with. The user flow gives a nice bird's-eye view of your software.
1.2 - For a non-expert on work shifts, I find it hard to understand the labels of the fields. For example, "Cluster Free Days" - "Number of Weeks to Cycle Over," etc. Can you please include a document explaining these fields?
1.3 - The input fields can only be controlled by the arrows, for example, _shift lengths [h]_ can only go up xx.33 but not for example xx.66. Is this on purpose to only use .33?
1.4 - I see "Load Combos" but "generate combinations", is combos, combinations? If so, kindly use a standardized language across.
1.5 - When I input a great workload, I get the message "10 weeks might require an extensive amount of computing power...", and the software becomes "non-responding"; as you can see my computer CPU and RAM are quite relaxed. I am not sure why that happens. Can you please elaborate on why is this happening although there are available computing resources? 
<img width="1134" alt="Capture" src="https://user-images.githubusercontent.com/63347261/149154150-60069f86-5a2c-41ed-9ab4-111d4d1e3c2a.PNG">

I will probably come with more questions after seeing the user flow diagrams since I could not navigate and understand the full software with just images and the caption.


2. Functionality documentation: Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
2.1 - I am not sure I find this anywhere. Can you please point me to it?
2.2 - To add, can you please at least provide Level 1 and 2 diagrams from the [C4 model ](https://c4model.com/) to understand better the purpose of the functions and classes you have created?


3. Substantial scholarly effort: Does this submission meet the scope eligibility described in the JOSS guidelines

This question goes back to @majensen as he is the moderator.

Specifically to these 2 points from the JOSS author guidelines:

1. "Your software should be a significant contribution to the available open-source software that either enables some new research challenges to be addressed or makes addressing research challenges significantly better (e.g., faster, easier, simpler).."

2. "Whether the software is sufficiently useful that it is likely to be cited by your peer group."

I am not sure there is a specific research challenge that has been addressed per se. 
As stated by the author, this is a  "Multi-Shift Rotational Workforce Schedules" generator. This is more of a nice to have "App" to me. I can find tons of them [online](https://peoplemanagingpeople.com/tools/best-employee-shift-scheduling-software/). So although this can be a good "app", it is not necessarily a research app. This brings me to the second point stated by the guidelines, and I think the answer to the next question is "No". But @benjaminbolling I am happy to hear your comments about this part and if you can highlight the research parts of the app? Thank you.

**Comment by benjaminbolling** [comment]
Thanks for your comment @magedhelmy1, a lot of things to address I see. :)

For point 3:
So the difference between this software and many apps available online is that this is a semi-automised method for generating shift patterns, whilst the ones shown in the list above allows more-or-less users (schedule-designers/makes) to insert and plan manually.

The reason why I would consider this as a research software is that it uses a computational approach to construct initial rotational shift work schedule patterns, and allows for implementation of further algorithms that can be developed and inserted (to be invoked e.g. under ` def createAllShiftPossibilities` ). Hence my assumption that it can address the research challenge of finding appropriate shift patterns equal for all employees.

Please let me know if this clarifies @magedhelmy1 , or if this maybe is not considered as a research app - to be clarified by @majensen .

I will address points 1+2 with their respective sub-points later today.

**Comment by magedhelmy1** [comment]
Hi Benjamin, Thanks for the reply.

The sentence you wrote under point 3 above should be reflected in the State-of-the-art part in your paper.
As the guidelines of JOSS state.

"..enables some new research challenges to be addressed or makes addressing research challenges significantly better (e.g., faster, easier, simpler).."

So kindly update the paper to clearly reflect this point and then you can cite those other software collections. Just a suggestion, maybe a simple table of the other software name, what they do, and how your software does it better :) But you are the author so it is up to you.

**Comment by majensen** [comment]
@magedhelmy1 thanks for picking this up again. You raise a fair point, and I would like to see @benjaminbolling address them with some specifics in the paper. I will then take a look, and may ask some other editors for their input. 

One approach, @benjaminbolling, may actually be to talk about your specific application in your work-- maintenance of scientific megainstruments round the clock so experiments don't fail. Can you highlight any specific problems that your software solves in this regard that are different from run-of-the-mill scheduling?

**Comment by benjaminbolling** [comment]
Hi, quick update - I had a baby recently and was hence very busy outside work. I have in the meanwhile done some fixes here and there, and expect to have things ready by the end of the WE

**Comment by majensen** [comment]
Congratulations @benjaminbolling ! Thanks for the update and try to get some sleep...

**Comment by majensen** [comment]
@benjaminbolling are you ready to restart the revision process? Thanks

**Comment by benjaminbolling** [comment]
yes, sorry - I could not imagine it would be so much to do with a baby. Combine that with a virus (a really long-lasting one, Covid-19 test showed negative though). Now we have all recovered from it and I will have time to finalize.

Is any of the reviewers available? I will take into account the last comments and update the text

**Comment by majensen** [comment]
Go ahead with your update @benjaminbolling - there is enough info in the review issue that I can proceed.

**Comment by majensen** [comment]
@benjaminbolling: "I could not imagine it would be so much to do with a baby" - I think if we knew beforehand, humans would be long extinct...

**Comment by arfon** [comment]
:wave: folks. Just checking in here. What's next to move this submission forward?

**Comment by majensen** [comment]
@editorialbot commands

**Comment by majensen** [comment]
@editorialbot remind me in 3 weeks

**Comment by majensen** [comment]
@editorialbot remind @benjaminbolling in 3 weeks

**Comment by benjaminbolling** [comment]
Hi again, so sorry for the humongous delay. Are the reviewers ok to do a final review? If yes, I can spend the rest of this week and most of next week on pushing the needed changes.

Otherwise, I would accept a withdraw of the submission.

**Comment by magedhelmy1** [comment]
Hi, I am not available to review this submission. I withdraw as a reviewer.

**Comment by majensen** [comment]
@editorialbot remove @magedhelmy1 as reviewer

**Comment by majensen** [comment]
@benjaminbolling do your best to complete by this Friday. I will look at it and see if I can get another pair of eyes who can take into account maged's comments. Thanks

**Comment by benjaminbolling** [comment]
Points 1.1-1.4 have been resolved in commit https://github.com/benjaminbolling/RSW/commit/cd615cbcbb274e3f5cf59c6ccac8bd704f6ff86c.

Point 1.5 will be solved by making some optimisation in the code, that will unfortunately need a little bit more time (which I am happy to state I will have in the coming weeks).

**Comment by benjaminbolling** [comment]
Point 3 hopefully considered as resolved in commit https://github.com/benjaminbolling/RSW/commit/cd615cbcbb274e3f5cf59c6ccac8bd704f6ff86c.

**Comment by benjaminbolling** [comment]
Point 2 hopefully resolved in https://github.com/benjaminbolling/RSW/commit/1ee4c61e9a8a3f8d4ea3c09adde8eb3174da6863.

**Comment by benjaminbolling** [comment]
@majensen : Up to you how you want to proceed now. In order to not cause more unnecessary work to anyone, I will make sure to address any comments the same day they pop up (putting a priority notifier for my Git emails).

**Comment by benjaminbolling** [comment]
Also, again, many thanks!

**Comment by benjaminbolling** [comment]
@whedon generate pdf

**Comment by benjaminbolling** [comment]
@editorialbot generate pdf

**Comment by majensen** [comment]
Hi @benjaminbolling - thanks for putting in this work. I think it is looking good. Can you please consider the following small edits:

l. 31: "Conclusively, " - delete
l. 69: "with" -> "to have"
l. 69: "n_{cf}" didn't get formatted (maybe an unusual underline character?)
l. 83: remove the colons (i.e., say 1s, 0s, or 'ones', 'zeroes')
l. 84: No need to cite a reference the common expression for 'n choose r' - you can remove this
l. 151: Citation should look like "In 2004, Burke (2004) made ...". This can be done with Latex, but can't remember how off hand.
l. 157 Citation has double parentheses - should be single
l. 162 (Petrovic & Berghe, 2002) -> Petrovic & Berghe (2002)
l. 162 "includes and" - delete
l. 164 (Laporte & Pesant, 2004)  -> Laporte & Pesant (2004)
l. 168-169 "does not give the user ... middle" -> "does not allow the user to perform the middle step"

**Comment by majensen** [comment]
@benjaminbolling - The associate editor-in-chief may think the paper as a whole is too long. Would you be willing to put the Benchmarking section in a readable form (markdown, eg) on your GitHub repo, and refer out to it in your paper?

**Comment by benjaminbolling** [comment]
@majensen Thank you for the comments, all your comments have been taken into account and implemented.

**Comment by majensen** [comment]
@editorialbot generate pdf

**Comment by majensen** [comment]
@editorialbot generate pdf

**Comment by majensen** [comment]
:wave: @openjournals/dev  - can you please have a look at the error thrown while generating pdf at https://github.com/openjournals/joss-reviews/issues/3431#issuecomment-1418302724 ? Thanks

**Comment by xuanxu** [comment]
The error is thrown because the author has no affiliation

**Comment by benjaminbolling** [comment]
I thought ORCID was sufficient. I will add this immediately.

**Comment by benjaminbolling** [comment]
@editorialbot generate pdf

**Comment by benjaminbolling** [comment]
I forgot; thanks @xuanxu for checking and pointing this out!

**Comment by majensen** [comment]
@magedhelmy1 I'm going to add you back as a reviewer here so that the automation recognizes you - no further work on your part is needed. I really appreciate your help on this paper. --@majensen

**Comment by majensen** [comment]
@benjaminbolling can you create a Zenodo archive of the code and return the DOI in this thread pls? Thanks

**Comment by benjaminbolling** [comment]
Done: https://doi.org/10.5281/zenodo.7618915

**Comment by majensen** [comment]
@editorialbot set 1.0.0 as version

**Comment by majensen** [comment]
@editorialbot set 10.5281/zenodo.7618915 as doi

**Comment by majensen** [comment]
@editorialbot set 10.5281/zenodo.7618915 as archive

**Comment by majensen** [comment]
@editorialbot recommend-accept

**Comment by majensen** [comment]
@benjaminbolling could you look at https://github.com/openjournals/joss-reviews/issues/3431#issuecomment-1423136022 and see if the reference recommedation makes sense , and if so add this to the bib?

**Comment by majensen** [comment]
Note @benjaminbolling is updating references per editorialbot https://github.com/openjournals/joss-reviews/issues/3431#issuecomment-1423136022
> 👋 @openjournals/sbcs-eics, this paper is ready to be accepted and published.
> 
> Check final proof 👉📄 [Download article](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03431/joss.03431/10.21105.joss.03431.pdf)
> 
> If the paper PDF and the deposit XML files look good in [openjournals/joss-papers#3953](https://github.com/openjournals/joss-papers/pull/3953), then you can now move forward with accepting the submission by compiling again with the command `@editorialbot accept`

**Comment by benjaminbolling** [comment]
Updated paper.bib for https://github.com/openjournals/joss-reviews/issues/3431#issuecomment-1423136022, see commit https://github.com/benjaminbolling/RSW/commit/91b09d016071baf422f3f793e164359269570bd0.

**Comment by majensen** [comment]
@editorialbot generate pdf

**Comment by majensen** [comment]
@editorialbot check references

**Comment by majensen** [comment]
@editorialbot recommend-accept

**Comment by arfon** [comment]
@majensen – I see a few unchecked boxes on this review still from reviewers. Is it fair to say the reviews are complete (and have recommended publication)?

**Comment by majensen** [comment]
Yes @arfon - they have been at it for almost two years and we have some reviewer fatigue. I am satisfied with the author's responsiveness during active review, and his work regarding functionality, documentation, and the addition of references. I think it will difficult to get the reviewers back around to dot the i's.

**Comment by arfon** [comment]
Got it. Thanks @majensen !

**Comment by arfon** [comment]
@benjaminbolling – I found a minor issue with your paper. Could you please merge this PR? https://github.com/benjaminbolling/RSW/pull/10

**Comment by benjaminbolling** [comment]
Good catch, thanks!

**Comment by arfon** [comment]
@editorialbot recommend-accept

**Comment by arfon** [comment]
@editorialbot accept

**Comment by arfon** [comment]
@ShantanuDash, @magedhelmy1 – many thanks for your reviews here and to @majensen  for editing this submission! JOSS relies upon the volunteer effort of people like you and we simply wouldn't be able to do this without you ✨ 

@benjaminbolling  – your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by benjaminbolling** [comment]
Many thanks all, especially to my reviewers @magedhelmy1 and @ShantanuDash and editor @majensen. Sorry for the long duration and for causing the fatigue, I appreciate all the effort and help.
